src/witan/workspace_api/utils.clj

```diff
@@ -1,6 +1,15 @@
 (ns witan.workspace-api.utils)
 
-(defmacro map-meta
+(defn assert!
+  [p msg]
+  (when-not (p) (throw (Exception. msg))))
```

Comment:
I'd prefer this (works with -> and ->> if used with partial too), avoids the name clash and doesn't appear to change state (the ! ) when it doesn't.

```
(defn property-holds? [p msg x]
  (if (p x)
    x
    (throw (Exception. msg))))

user> (property-holds? odd? "Not odd" 1) 
1
user> (property-holds? odd? "Not odd" 2) 
Exception Not odd  sun.reflect.NativeConstructorAccessorImpl.newInstance0 (NativeConstructorAccessorImpl.java:-2)
```